### PR TITLE
fix cmake for tests

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- fix path to internal `catch2`
- remove keyword to `general` (it is optional and the default, was also used with the wrong syntax)

@BenjaminW3 Why do we linked against `general`. I do not find any information on what kind of library it is.

### CODE REVIEW

**Comment by BenjaminW3** [comment]
general is no lib but a [CMake keyword](https://cmake.org/cmake/help/v3.0/command/target_link_libraries.html). It is purely optional so it can be removed.

File: test/common/CMakeLists.txt [hunk 1]
```diff
@@ -53,7 +53,7 @@ TARGET_COMPILE_OPTIONS(
     PUBLIC ${_COMMON_COMPILE_OPTIONS_PUBLIC})
 TARGET_LINK_LIBRARIES(
     ${_COMMON_TARGET_NAME}
-    PUBLIC alpaka ${_COMMON_LINK_LIBRARIES_PUBLIC})
+    PUBLIC "alpaka;${_COMMON_LINK_LIBRARIES_PUBLIC}")
```
Comment (psychocoderHPC) [inline test/common/CMakeLists.txt]: I looked again into the [cmake documentation](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) and it looks like this must be separated by spaces.
I will change it.

**Comment by psychocoderHPC** [comment]
> general is no lib but a [CMake keyword](https://cmake.org/cmake/help/v3.0/command/target_link_libraries.html). It is purely optional so it can be removed.

Ahh the syntax was wrong. If you like to use `general` you must have a space before the library name `    LIST(APPEND _COMMON_LINK_LIBRARIES_PUBLIC "general ${CUDA_CUDA_LIBRARY}")`

File: test/common/CMakeLists.txt (hunk 1 above)
Comment (psychocoderHPC) [reply test/common/CMakeLists.txt]: CMake is allowing a lists and space separated entries. Never the less I will change it to separate entries.

**Review by BenjaminW3 — APPROVED** [review]

**Pull request merged**